src/components/Search.js

```diff
@@ -0,0 +1,30 @@
+import React, { useState } from "react";
+function Search(props) {
+
+    const [search, setSearch] = useState("");
+
+    const filter = (searchedFood) => {
+        if (!searchedFood) {
+            props.setFoods(props.foods)
+        } else {
+            props.setFoods((prevValue) => {
+                return prevValue.filter(food => food.name.toLowerCase().includes(searchedFood.toLowerCase()));
+            });
+        }
+    }
```

Comment:
If you write your filter (search) function in this component, you must have a state that hold the value of originla array -> otherwise we have no access to it.

Better approach would be to store this function in app.js -> and pass it down to Search.js  (call it from search with the searc input value)